core/types/block.go

```diff
@@ -90,6 +90,10 @@ type Header struct {
 	LastCommitSignature [96]byte    `json:"lastCommitSignature"  gencodec:"required"`
 	LastCommitBitmap    []byte      `json:"lastCommitBitmap"     gencodec:"required"` // Contains which validator signed
 	ShardStateHash      common.Hash `json:"shardStateRoot"`
+	Vrf                 [32]byte    `json:"vrf"`
```

Comment:
We should test whether the addition of empty new []bytes change the hash of the block. If not, I think we may not need extra versioning. Otherwise, we need to use different version of the struct for the hard fork.

Reply:
This comes admittedly a bit too late, but I must point out that this does change the hash: Because this adds 32 bytes of data.  Even if we use a byte slice, an empty slice consists of one header+length indicator byte, so it is not completely backward compatible.

We may need to revert this PR until we have a proper data structure versioning strategy.